src/TesApi.Web/AzureProxy.cs

```diff
@@ -430,19 +561,34 @@ public async Task<IEnumerable<string>> GetActivePoolIdsAsync(string prefix, Time
             return (await batchClient.PoolOperations.ListPools(activePoolsFilter).ToListAsync(cancellationToken)).Select(p => p.Id);
         }
 
+        /// <inheritdoc/>
+        public IAsyncEnumerable<CloudPool> GetActivePoolsAsync(string hostName)
+        {
+            var activePoolsFilter = new ODATADetailLevel
+            {
+                FilterClause = $"state eq 'active'",
+                SelectClause = BatchPool.CloudPoolSelectClause
+            };
+
+            return batchClient.PoolOperations.ListPools(activePoolsFilter).ToAsyncEnumerable()
+                .Where(p => hostName.Equals(p.Metadata?.FirstOrDefault(m => BatchScheduler.PoolHostName.Equals(m.Name, StringComparison.Ordinal))?.Value, StringComparison.OrdinalIgnoreCase));
```

Comment:
The metadata name is searched case sensitive to avoid issues with other entities adding metadata that might differ by case (assuming that's possible). The value is compared case insensitive. The difference is deliberate.